### PR TITLE
nit: close modal when user clicks outside content

### LINKED ISSUE #920: Nicer UX for exiting network/asset selection popups
It's a little annoying to have to click the "x" in the top right of the screen when selecting network/asset in connect. The UX I'd expect is being able to exit the selection popup by clicking anywhere outside of the main selection UI (so anywhere in the dark background).

For example, in the below screen for selecting a network, a user can only exit the selection popup by clicking the "x" in the top right. Most UIs allow users to exit popups by clicking anywhere outside of the popup content.

connect_13

### PR DESCRIPTION
**Pull request by anondev2323**
Quick fix for #920

### CODE REVIEW

File: wormhole-connect/src/components/Modal.tsx [hunk 1]
```diff
@@ -61,10 +61,16 @@ function Modal({ open, width, closable, children, onClose }: Props) {
...
+      onClose={(event, reason) => {
+        console.log(reason);
+        if (reason === 'backdropClick') {
+          onClose();
+        }
+      }}
       sx={{ borderRadius: 8 }}
       fullWidth
       fullScreen
+      onClick={onClose}
```
Comment (evan-gray) [inline wormhole-connect/src/components/Modal.tsx]: onClose is fired on click?

File: wormhole-connect/src/components/Modal.tsx [hunk 2]
```diff
@@ -61,10 +61,16 @@ function Modal({ open, width, closable, children, onClose }: Props) {
   return (
     <Dialog
       open={open}
-      onClose={onClose}
+      onClose={(event, reason) => {
+        console.log(reason);
```
Comment (evan-gray) [inline wormhole-connect/src/components/Modal.tsx]: Probably can remove this console log

File: wormhole-connect/src/components/Modal.tsx [hunk 3]
```diff
@@ -61,10 +61,16 @@ function Modal({ open, width, closable, children, onClose }: Props) {
   return (
     <Dialog
       open={open}
-      onClose={onClose}
+      onClose={(event, reason) => {
```
Comment (evan-gray) [inline wormhole-connect/src/components/Modal.tsx]: Might want to consider use callback to avoid re-render / re-attaching the event every render.

**Review by evan-gray — CHANGES_REQUESTED** [review]
Is it worth considering simply removing `fullScreen`?

File: wormhole-connect/src/components/Modal.tsx (hunk 1 above)
Comment (evan-gray) [reply wormhole-connect/src/components/Modal.tsx]: @anondev2323 confirmed, this causes the dialog to close even when you click on, like, the search box or the heading, etc.

File: wormhole-connect/src/components/Modal.tsx (hunk 3 above)
Comment (evan-gray) [reply wormhole-connect/src/components/Modal.tsx]: This event never seems to fire for me (after removing the onClick)

**Review by evan-gray — DISMISSED** [review]
Made changes to prevent the onClick from closing when clicking inside the content. Also fixed a double-scrollbar issue.

File: wormhole-connect/src/components/Modal.tsx [hunk 4]
```diff
@@ -26,6 +26,8 @@ const useStyles = makeStyles<{ width: number }>()((theme: any, { width }) => ({
       margin: '20px auto',
       padding: '24px 12px',
     },
+    background: 'rgba(255,255,255,.02)',
```
Comment (anondev2323) [inline wormhole-connect/src/components/Modal.tsx]: Maybe it's related to this line.  But I noticed an ever so slight difference in color between the inner/outter modal content.  Should all be the same color

Comment (evan-gray) [reply wormhole-connect/src/components/Modal.tsx]: ah, did that to discern the difference (of when clicking would close vs not), but agree with leaving as is for now. can always revisit.

**Review by evan-gray — APPROVED** [review]

**Pull request merged**